### PR TITLE
feat(github-actions): enable various imperative github actions

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction :pencil2:
Ensure various imperative pipelines are available for consistent pipeline actions across all the applications.

## Resolution :heavy_check_mark:
* Added `github.yml` for PR lint check.
* Added `prettier:all` check to `lint.yml`.
* Added `pr.yml` check for stale PR validation.
* Updated `publish.yml` as per latest `Google Release please` release notes.

## Miscellaneous :heavy_plus_sign:
* Dependencies updates
* Moved GHA hardcoded variables to Enterprise scope GH secrets / variable
* Refactored GitHooks

### CODE REVIEW

File: .github/workflows/deployment.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ on:
 env:
   PRODUCT: apim
   ENVIRONMENT: ${{ github.ref_name }}
-  TIMEZONE: "Europe/London"
+  timezone: ${{ vars.TIMEZONE }}
```
Comment (oscar-richardson-softwire) [inline .github/workflows/deployment.yml:42]: Why the change to lowercase?

File: .github/workflows/infrastructure.yml [hunk 2]
```diff
@@ -68,7 +68,7 @@ on:
 env:
   PRODUCT: apim
   ENVIRONMENT: infrastructure
-  TIMEZONE: "Europe/London"
+  timezone: ${{ vars.TIMEZONE }}
```
Comment (oscar-richardson-softwire) [inline .github/workflows/infrastructure.yml:71]: Why the change to lowercase?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -16,8 +16,7 @@ on:
 
 env:
   environment: "qa"
-  timezone: "Europe/London"
-  node: ${{ vars.NODE_VERSION }}
```
Comment (oscar-richardson-softwire) [inline .github/workflows/test.yml:20]: Just checking that we no longer need the `node` env var here

File: .github/workflows/github.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
+name: GitHub
```
Comment (oscar-richardson-softwire) [inline .github/workflows/github.yml:1]: Could we have a comment to explain the purpose of this GHA (like the others do) please?

File: .github/workflows/github.yml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+run-name: 🔬Pull request inspection on ${{ github.event.number }}
```
Comment (oscar-richardson-softwire) [inline .github/workflows/github.yml:2]:
```suggestion
run-name: 🔬 Pull request inspection on ${{ github.event.number }}
```

File: .github/workflows/pr.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
+# This YAML configuration defines a GitHub Actions workflow named "PR" designed to analyse pull requests
+# for staleness and take action accordingly.
+#
+# Identify: Uses the actions/stale@v5 action with parameters to mark pull requests as stale if
+# they have been inactive for a certain number of days and close them if they exceed another threshold.
+# This workflow helps maintain the health of the pull request queue by identifying and managing stale pull requests automatically.
+#
+
+name: PR
+run-name: 🔎 pull request analysis on ${{ github.event.number }}
```
Comment (oscar-richardson-softwire) [inline .github/workflows/pr.yml:10]:
```suggestion
run-name: 🔎 Pull request analysis on ${{ github.event.number }}
```

File: .github/workflows/publish.yml [hunk 7]
```diff
@@ -1,6 +1,11 @@
-# MDM-API Git Hub Actions
+# MDM Git Hub Actions
 #####################################
-# This GHA is responsible for release version updates.
+# This GHA is responsible for creating automated
+# GitHub release based PR. Once the PR is merged
+# Google release please will update the `CHANGELOG.md`
+# `README.md` and `package.json` files.
+# Followed by tagging the commit with the appropriate version
+# number (as per SemVer) and create a GitHub release on the tag.
```
Comment (oscar-richardson-softwire) [inline .github/workflows/publish.yml:8]:
```suggestion
# This GHA is responsible for creating automated
# GitHub release-based PRs. Once the PR is merged,
# Google Release Please will update the `CHANGELOG.md`
# `README.md` and `package.json` files, followed by tagging 
# the commit with the appropriate version number (as per SemVer)
# and creating a GitHub release on the tag.
```

**Review by oscar-richardson-softwire — COMMENTED** [review]
Just a few comments :)

File: .github/workflows/deployment.yml (hunk 1 above)
Comment (abhi-markan) [reply .github/workflows/deployment.yml:42]: This is to keep it consistent with other repositories.

File: .github/workflows/github.yml (hunk 4 above)
Comment (abhi-markan) [reply .github/workflows/github.yml:1]: Have added.

File: .github/workflows/github.yml [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    steps:
+      - name: Environment 🧪
+        run: echo "Environment set to ${{ env.environment }}"
+
+      - name: Timezone 🌐
+        run: echo "Timezone set to ${{ env.timezone }}"
+
+  # 2. Pull request
+  pullrequest:
+    name: Pull request ⬇️
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply .github/workflows/github.yml:31]: Good spot have added.

File: .github/workflows/infrastructure.yml (hunk 2 above)
Comment (abhi-markan) [reply .github/workflows/infrastructure.yml:71]: Same as above.

File: .github/workflows/publish.yml (hunk 7 above)
Comment (abhi-markan) [reply .github/workflows/publish.yml:8]: Have updated.

File: .github/workflows/test.yml (hunk 3 above)
Comment (abhi-markan) [reply .github/workflows/test.yml:20]: Nope they have reffered directly now.

File: .prettierignore [hunk 9]
```diff
@@ -1,6 +1,19 @@
+# Root
+node_modules/
 dist/
-.cache/
+build/
+public/
+generated_reports/
+coverage/
+package*.json
+.cache
```
Comment (oscar-richardson-softwire) [inline .prettierignore:9]:
```suggestion
.cache/
```

File: .prettierignore [hunk 10]
```diff
@@ -1,6 +1,19 @@
...
+
+# Sub-directories
+**/node_modules/
+**/dist/
+**/build/
+**/public/
+**/**/**/generated_reports/
```
Comment (oscar-richardson-softwire) [inline .prettierignore:16]: Why `**/**/**`/ not `**/`?

File: .github/workflows/github.yml [hunk 11]
```diff
@@ -1,14 +1,18 @@
...
 name: GitHub
-run-name: 🔬Pull request inspection on ${{ github.event.number }}
+run-name: 🔬 Pull request inspection on ${{ github.event.number }}
 
 on:
   pull_request:
-    branches:
-      - main
-      - "release-*"
+    branches: [main, main-*]
```
Comment (oscar-richardson-softwire) [inline .github/workflows/github.yml:12]: Why the change from `release-*` to `main-*`?

File: .github/workflows/lint.yml [hunk 12]
```diff
@@ -1,9 +1,37 @@
...
+# Lint:
+# Name: Scanning
+# Depends on: setup
+# Runs on: ubuntu-latest
+# Steps:
+# Sets the timezone using the specified value.
+# Checks out the repository code.
+# Sets up the specified Node.js version.
+# Installs dependencies using npm run ci:all.
+# Runs linting using the command npm run lint:all.
```
Comment (oscar-richardson-softwire) [inline .github/workflows/lint.yml:24]: Should we add docs for step 3 (i.e., the Prettier step?)

File: .github/workflows/github.yml (hunk 11 above)
Comment (abhi-markan) [reply .github/workflows/github.yml:12]: We only use `main-*` branches in `APIM` not `release-*`

File: .prettierignore (hunk 10 above)
Comment (abhi-markan) [reply .prettierignore:16]: Have amended

File: .prettierignore (hunk 9 above)
Comment (abhi-markan) [reply .prettierignore:9]: This is a file

File: .github/workflows/lint.yml [hunk 13]
```diff
@@ -59,3 +97,31 @@ jobs:
       - name: Linting
         working-directory: ./
         run: npm run lint
+
+  # 3. Lint
```
Comment (oscar-richardson-softwire) [inline .github/workflows/lint.yml:101]:
```suggestion
  # 3. Prettier
```

Comment (abhi-markan) [reply .github/workflows/lint.yml]: Updated, good spot.

**Review by oscar-richardson-softwire — APPROVED** [review]
LGTM :)
